tracing-journald/src/lib.rs

```diff
@@ -0,0 +1,241 @@
+            prefix: self.field_prefix.as_ref().map(|x| &x[..]),
+        });
+
+        span.extensions_mut().insert(SpanFields(buf));
+    }
+
+    fn on_record(&self, id: &Id, values: &Record, ctx: Context<S>) {
+        let span = ctx.span(id).expect("unknown span");
+        let depth = span.parents().count();
+        let mut exts = span.extensions_mut();
+        let buf = &mut exts.get_mut::<SpanFields>().expect("missing fields").0;
+        values.record(&mut SpanVisitor {
+            buf,
+            depth,
+            prefix: self.field_prefix.as_ref().map(|x| &x[..]),
+        });
+    }
+
+    fn on_event(&self, event: &Event, ctx: Context<S>) {
+        let mut buf = Vec::with_capacity(256);
+
+        // Record span fields
+        for span in ctx.scope() {
+            let exts = span.extensions();
+            let fields = exts.get::<SpanFields>().expect("missing fields");
+            buf.extend_from_slice(&fields.0);
+        }
+
+        // Record event fields
+        put_metadata(&mut buf, event.metadata(), None);
+        event.record(&mut EventVisitor::new(
+            &mut buf,
+            self.field_prefix.as_ref().map(|x| &x[..]),
+        ));
+
+        // What could we possibly do on error?
+        #[cfg(unix)]
+        let _ = self.socket.send(&buf);
+    }
+}
+
+struct SpanFields(Vec<u8>);
+
+struct SpanVisitor<'a> {
+    buf: &'a mut Vec<u8>,
+    depth: usize,
+    prefix: Option<&'a str>,
+}
+
+impl Visit for SpanVisitor<'_> {
+    fn record_debug(&mut self, field: &Field, value: &dyn fmt::Debug) {
+        write!(self.buf, "S{}", self.depth).unwrap();
+        if let Some(prefix) = self.prefix {
+            self.buf.extend_from_slice(prefix.as_bytes());
+        }
+        self.buf.push(b'_');
+        put_debug(self.buf, field.name(), value);
+    }
+}
+
+/// Helper for generating the journal export format, which is consumed by journald:
+/// https://www.freedesktop.org/wiki/Software/systemd/export/
+struct EventVisitor<'a> {
+    buf: &'a mut Vec<u8>,
+    prefix: Option<&'a str>,
+}
+
+impl<'a> EventVisitor<'a> {
+    fn new(buf: &'a mut Vec<u8>, prefix: Option<&'a str>) -> Self {
+        Self { buf, prefix }
+    }
+}
+
+impl Visit for EventVisitor<'_> {
+    fn record_debug(&mut self, field: &Field, value: &dyn fmt::Debug) {
+        if let Some(prefix) = self.prefix {
+            if field.name() != "message" {
+                self.buf.extend_from_slice(prefix.as_bytes());
+                self.buf.push(b'_');
+            }
+        }
+        put_debug(self.buf, field.name(), value);
+    }
+}
+
+fn put_metadata(buf: &mut Vec<u8>, meta: &Metadata, span: Option<usize>) {
+    if span.is_none() {
+        put_field(
+            buf,
+            "PRIORITY",
+            match *meta.level() {
+                Level::ERROR => b"3",
+                Level::WARN => b"4",
+                Level::INFO => b"5",
+                Level::DEBUG => b"6",
+                Level::TRACE => b"7",
+            },
```

Comment:
I think maybe the docs should document how `tracing` levels are mapped to journald priorities?